From 5ac2f554152d3d0246d95f9851571b2ff7c3ce0a Mon Sep 17 00:00:00 2001
From: Ranbir Singh <Ranbir.Singh3@Dell.com>
Date: Tue, 24 Jan 2023 11:53:17 +0530
Subject: [PATCH 2/2] NetworkPkg/HttpDxe: Fix DEADCODE Coverity issue

The function HttpResponseWorker does an assignment -

  StatusCodeStr = HttpHeaders + AsciiStrLen (HTTP_VERSION_STR) + 1;

and the result of pointer arithmetic

  "HttpHeaders + AsciiStrLen("HTTP/1.1") + 1" is never null.

Hence at condition check if (StatusCodeStr == NULL), the value of
"StatusCodeStr" cannot be "NULL" and the check can be removed.

Cc: Maciej Rabeda <maciej.rabeda@linux.intel.com>
Cc: Jiaxin Wu <jiaxin.wu@intel.com>
Cc: Siyuan Fu <siyuan.fu@intel.com>
REF: https://bugzilla.tianocore.org/show_bug.cgi?id=4252
Signed-off-by: Ranbir Singh <Ranbir.Singh3@Dell.com>
---
 NetworkPkg/HttpDxe/HttpImpl.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/NetworkPkg/HttpDxe/HttpImpl.c b/NetworkPkg/HttpDxe/HttpImpl.c
index 7c5c925cf78b..6e70759944aa 100644
--- a/NetworkPkg/HttpDxe/HttpImpl.c
+++ b/NetworkPkg/HttpDxe/HttpImpl.c
@@ -1117,11 +1117,6 @@ HttpResponseWorker (
     // Search for Status Code.
     //
     StatusCodeStr = HttpHeaders + AsciiStrLen (HTTP_VERSION_STR) + 1;
-    if (StatusCodeStr == NULL) {
-      Status = EFI_NOT_READY;
-      goto Error;
-    }
-
     StatusCode = AsciiStrDecimalToUintn (StatusCodeStr);
 
     //
-- 
2.36.1.windows.1

